presto-thrift/src/main/java/com/facebook/presto/thrift/ThriftConnector.java

```diff
@@ -0,0 +1,113 @@
+import io.airlift.log.Logger;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class ThriftConnector
+        implements Connector
+{
+    private static final Logger log = Logger.get(ThriftConnector.class);
+    private final LifeCycleManager lifeCycleManager;
+    private final ThriftMetadata metadata;
+    private final ThriftSplitManager splitManager;
+    private final ThriftPageSourceProvider pageSourceProvider;
+    private final ThriftInternalSessionProperties internalSessionProperties;
+    private final ThriftClientSessionProperties clientSessionProperties;
+    private final ThriftIndexProvider indexProvider;
+
+    @Inject
+    public ThriftConnector(LifeCycleManager lifeCycleManager,
+            ThriftMetadata metadata,
+            ThriftSplitManager splitManager,
+            ThriftPageSourceProvider pageSourceProvider,
+            ThriftInternalSessionProperties internalSessionProperties,
+            ThriftClientSessionProperties clientSessionProperties,
+            ThriftIndexProvider indexProvider)
+    {
+        this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
+        this.internalSessionProperties = requireNonNull(internalSessionProperties, "internalSessionProperties is null");
+        this.clientSessionProperties = requireNonNull(clientSessionProperties, "clientSessionProperties is null");
+        this.indexProvider = requireNonNull(indexProvider, "indexProvider is null");
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+    {
+        return ThriftTransactionHandle.INSTANCE;
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata(ConnectorTransactionHandle transactionHandle)
+    {
+        return metadata;
+    }
+
+    @Override
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    @Override
+    public ConnectorPageSourceProvider getPageSourceProvider()
+    {
+        return pageSourceProvider;
+    }
+
+    @Override
+    public List<PropertyMetadata<?>> getSessionProperties()
+    {
+        return ImmutableList.<PropertyMetadata<?>>builder()
+                .addAll(internalSessionProperties.getSessionProperties())
+                .addAll(clientSessionProperties.getSessionProperties())
+                .build();
+    }
+
+    @Override
+    public ConnectorIndexProvider getIndexProvider()
+    {
+        return indexProvider;
+    }
+
+    @Override
+    public final void shutdown()
+    {
+        try {
+            lifeCycleManager.stop();
+        }
+        catch (Exception e) {
```

Comment:
Technically `lifeCycleManager.stop()` is allowed to throw interrupted exception. I can't claim that it is happening IRL. Also, since we do not support unload for the connectors, it doesn't matter a lot. So, this is more like a best practice suggestion.